iter_gen_homework/iterators.py

```diff
@@ -0,0 +1,82 @@
+class SimpleIterator:
+    def __init__(self, iterable):
+        self.iterable = iterable
+        self.index = 0
+
+    def __next__(self):
+        try:
+            val = self.iterable[self.index]
+            self.index += 1
+            return val
+        except Exception:
+            raise StopIteration
+
+    def __iter__(self):
+        return self
+
+
+class ReversedIterator:
+
+    def __init__(self, iterable):
+        self.iterable = iterable
+
+    def __next__(self):
+        ...
+
+    def __iter__(self):
+        yield from self.iterable[::-1]
```

Comment:
its not an iterator

Reply:
# iterator should support:
iterator = ReversedIterator(iterable)
next(iterator)
next(iterator)
next(iterator)
...